src/actions/vdom/add-attribute.js

```diff
@@ -0,0 +1,16 @@
+var toCamelCase = require('../../utils/to-camel-case');
+
+module.exports = function addAttribute(node, key, val) {
+
+  node = (node.vNode || node);
+  node.properties.dataset = (node.properties.dataset || {});
+
+  if (/data/.test(key)) {
```

Comment:
I think we might've had a chat about this earlier, but what was the reason for inserting into `dataset` sometimes and sometimes straight into `properties`? Probably worthy of a comment in the code.